app/scripts/controllers/console.js

```diff
@@ -283,9 +284,18 @@ angular.module('console', ['sql', 'datatypechecks'])
           lineWrapping: true
         });
 
-        if ($location.search().query) {
-          editor.setValue($location.search().query);
+        function updateStatementOnUrlSearch() {
+          if ($location.search().query) {
+            editor.setValue($location.search().query);
+            $console.updateStatement($location.search().query);
+          }
         }
+        updateStatementOnUrlSearch();
+
+        $scope.$on('$routeUpdate', function () {
```

Comment:
couldn't you just bind the function like `$scope.$on('$routeUpdate', updateStatementOnUrlSearch.bind(this));` ?